app/src/main/java/com/epfl/Utils/drawyourpath/Utils.kt

```diff
@@ -30,23 +35,125 @@ object Utils {
         return future
     }
 
+    /**
+     * get the profile photo as bitmap or if none the default profile photo
+     * @param photo the photo to decode
+     * @param res the resources used to get the default profile photo
+     * @return the profile photo or the default profile photo
+     */
+    fun decodePhotoOrGetDefault(photo: ByteArray?, res: Resources): Bitmap {
+        if (photo == null) {
+            return getDefaultPhoto(res)
+        }
+        return BitmapFactory.decodeByteArray(photo, 0, photo.size, BitmapFactory.Options()) ?: getDefaultPhoto(res)
+    }
+
+    /**
+     * Decodes the photo from base64 string to bitmap format
+     * @param photoStr photo encoded
+     * @return the photo in bitmap format
+     */
+    fun decodePhotoOrGetDefault(photoStr: String?, res: Resources): Bitmap {
+        return decodePhotoOrGetDefault(decodeStringAsByteArray(photoStr), res)
+    }
+
     /**
      * Decodes the photo from base64 string to bitmap format and return null if the dataSnapShot is null
      * @param photoStr photo encoded
      * @return the photo in bitmap format, and null if no photo is stored on the database
      */
     fun decodePhoto(photoStr: String): Bitmap {
-        val tabByte = Base64.getDecoder().decode(photoStr as String)
+        val tabByte = Base64.getDecoder().decode(photoStr)
         return BitmapFactory.decodeByteArray(tabByte, 0, tabByte.size)
     }

```

Comment:
There is code duplication. We should add 
```kotlin
fun decodePhoto(photo: ByteArray): Bitmap {
    return BitmapFactory.decodeByteArray(photo, 0, tabByte.size)
}
```
and use it in `decodePhoto(photoStr: String)` and `decodePhotoOrGetDefault(photo: ByteArray?, res: Resources): Bitmap`.

Also, `decodePhoto(photoStr: String)` should use `decodeStringAsByteArray()`.

Reply:
Thank you for pointing this out, I also changed the return value of the decodePhoto to `Bitmap?` as `BitmapFactory.decodeByteArray` can return null if it couldn't decode the photo.